cmake/external/abseil-cpp.cmake

```diff
@@ -38,3 +38,11 @@ if (GDK_PLATFORM)
   # tell Abseil to pretend we're building an APP.
   target_compile_definitions(absl_symbolize PRIVATE WINAPI_FAMILY=WINAPI_FAMILY_DESKTOP_APP)
 endif()
+
+if(NOT onnxruntime_DISABLE_ABSEIL)
+  set(ABSEIL_LIBS absl::inlined_vector absl::flat_hash_set
+    absl::flat_hash_map absl::node_hash_set absl::node_hash_map absl::base absl::throw_delegate absl::raw_hash_set
+    absl::hash absl::city absl::low_level_hash absl::raw_logging_internal)
+else()
+  add_compile_definitions(DISABLE_ABSEIL)
```

Comment:
given these lines:
https://github.com/microsoft/onnxruntime/blob/91dcc37782164e3bd295d8d695563a3a046a26f7/cmake/external/onnxruntime_external_deps.cmake#L22-L24
looks like this condition will never be hit and the symbol won't be defined.
maybe we should have a build which exercises `onnxruntime_DISABLE_ABSEIL=ON`